src/main/java/org/mockito/internal/invocation/InvocationMatcher.java

```diff
@@ -134,24 +133,12 @@ private void captureVarargsPart(Invocation invocation) {
             return;
         }
 
-        Object[] rawArguments = invocation.getRawArguments();
-        int indexOfVarArg = rawArguments.length - 1;
+        Object[] arguments = invocation.getArguments();
 
-        Object varArg = rawArguments[indexOfVarArg];
-        if (varArg == null) {
-            return;
-        }
-        int varArgSize = Array.getLength(varArg);
-
-        if (matchers.size() == indexOfVarArg) {
-            return;
-        }
-
-        for (int i = 0; i < varArgSize; i++) {
-            int indexOfVarargMatcher = (i + indexOfVarArg > matchers.size() -1 ? indexOfVarArg : i + indexOfVarArg);
-            ArgumentMatcher m = matchers.get(indexOfVarargMatcher);
+        for (int i = 0; i < arguments.length; i++) {
+            ArgumentMatcher m = matchers.get(i > matchers.size() - 1 ? matchers.size() - 1 : i);
             if (m instanceof CapturesArguments) {
-                ((CapturesArguments) m).captureFrom(Array.get(varArg, i));
+                ((CapturesArguments) m).captureFrom(arguments[i]);
             }
         }
```

Comment:
@ChristianSchwarz 
Hi Christian,

I debug-ed it, and it's using this null value to have a different type of invocation. If we enhance it to use `String[0]` instead of `String[] { null }`

Here is the code that will match this invocations:

```java
    public void check(List<Invocation> invocations, InvocationMatcher wanted) {
        List<Invocation> actualInvocations = finder.findInvocations(invocations, wanted);
        
        if (actualInvocations.isEmpty()) {
            Invocation similar = finder.findSimilarInvocation(invocations, wanted);
            if (similar != null) {
                ArgumentMatchingTool argumentMatchingTool = new ArgumentMatchingTool();
                Integer[] indexesOfSuspiciousArgs = argumentMatchingTool.getSuspiciouslyNotMatchingArgsIndexes(wanted.getMatchers(), similar.getArguments());
                SmartPrinter smartPrinter = new SmartPrinter(wanted, similar, indexesOfSuspiciousArgs);
                reporter.argumentsAreDifferent(smartPrinter.getWanted(), smartPrinter.getActual(), similar.getLocation());
            } else {
                reporter.wantedButNotInvoked(wanted, invocations);
            }
        }
    }
``` in `MissingInvocationChecker`.